lib/bundler/process_lock.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Bundler
+  class ProcessLock
+    def self.lock(bundle_path = Bundler.bundle_path)
+      lock_file_path = File.join(bundle_path, "bundler.lock")
+      File.open(lock_file_path, "w") do |f|
```

Comment:
On second thoughts @segiddins the delete could be an issue in the way I implemented it, you could have opened the file and awaiting a lock while someone else who had the lock has gone and deleted it from under you.